packages/entity-server/src/routes/hierarchy/relationships/MultiEntityRelationshipsRoute.ts

```diff
@@ -4,28 +4,11 @@
  */
 
 import { Route } from '@tupaia/server-boilerplate';
-import { MultiEntityRelationshipsRequest, RelationshipsResponseBody } from './types';
+import { MultiEntityRelationshipsRequest } from './types';
 import { ResponseBuilder } from './ResponseBuilder';
 
 export class MultiEntityRelationshipsRoute extends Route<MultiEntityRelationshipsRequest> {
   async buildResponse() {
-    const { entities } = this.req.ctx;
-
-    const allResponses: RelationshipsResponseBody = {};
-    await Promise.all(
-      entities.map(async entity => {
-        const singleCtx = { ...this.req.ctx, entity };
-        const response = await new ResponseBuilder(
-          this.req.models,
-          singleCtx,
-          this.req.query.groupBy,
-        ).build();
-        Object.entries(response).forEach(([key, value]) => {
-          allResponses[key] = value;
-        });
-      }),
-    );
-
-    return allResponses;
+    return new ResponseBuilder(this.req.models, this.req.ctx, this.req.query.groupBy).build();
```

Comment:
I should be fair about the performance, it does improve first time requests by a noticeable amount, however since we cache the results after that it's hard to notice the difference.

Worth noting though that even the worst performing countries in `PSSS` only have around 400 facilities, if we had to do this for all schools in Laos or something then we'd probably see more benefit from this re-write 👍